sys/net/gnrc/application_layer/static/gnrc_static.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/rpl.h"
+#include "net/ipv6/addr.h"
+#include "net/netdev.h"
+#include "net/netopt.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static void _config_upstream(gnrc_netif_t *upstream)
+{
+    ipv6_addr_t addr;
+
+    const char static_addr[] =
+#ifdef GNRC_IPV6_STATIC_ADDR
+        GNRC_IPV6_STATIC_ADDR;
+#else
+        NULL;
```

Comment:
Is that possible? Assigning `NULL` to a an array? Wouldn't it rather make sense to error out here?

```suggestion
#error "gnrc_static used but no GNRC_IPV6_STATIC_ADDR define provided";
```

or something like that.